File: LogginColombiaGold/frmPpal.cs
using System;
using System.Configuration;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace LogginColombiaGold
{
    public partial class frmPpal : Form
    {
        clsRf oRf = new clsRf();
        DataTable dtFormsAllowed = new DataTable();

        public frmPpal()
        {
            InitializeComponent();
        }

        private void logginToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            //clsRf.sIdGrupo = "3";
            //dtFormsAllowed = oRf.getFormsByGrupo(clsRf.sIdGrupo, ConfigurationSettings.AppSettings["IDProject"].ToString());
            //clsRf.dsPermisos = oRf.getFormsByGrupoAll(clsRf.sIdGrupo);
            try
            {
                //***** Este fragmento es utilizado para realización de modificaciones y pruebas. AAA ****

                //frmLoggin oLog = new frmLoggin();
                //oLog.MdiParent = this;
                //oLog.Show();

                //****************************************************************
                // Para pruebas se debe comentar el siguiente fragmento de codigo.

                DataRow[] dato = dtFormsAllowed.Select("nombre_Real_Form = 'frmLoggin'");
                if (dato.Length > 0)
                {
                    frmLoggin oLog = new frmLoggin();
                    oLog.MdiParent = this;
                    oLog.Show();
                }
                else
                {
                    MessageBox.Show("Form is not allowed", "Shipment", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error: " + ex.Message);
            }
        }

        private void frmPpal_Load(object sender, EventArgs e)
        {
            try
            {
                mnuValidate.Visible = true;
                mnuValidate.Enabled = true;

                dtFormsAllowed = oRf.getFormsByGrupo(clsRf.sIdGrupo, ConfigurationSettings.AppSettings["IDProject"].ToString());
                clsRf.dsPermisos = oRf.getFormsByGrupoAll(clsRf.sIdGrupo);

                MdiClient ctlMDI = default(MdiClient);
                foreach (Control ctl in this.Controls)
                {
                    try
                    {
                        ctlMDI = (MdiClient)ctl;
                        ctlMDI.BackColor = Color.White;
                    }
                    catch (InvalidCastException ex)
                    {
                    }
                }


            }
            catch (Exception ex)
            {
                MessageBox.Show("Error: " + ex.Message);
            }
        }

        private void logOutToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Application.Exit();
        }

        private void frmPpal_FormClosed(object sender, FormClosedEventArgs e)
        {
            clsRf.dsPermisos = new DataSet();
            Application.Exit();
        }

        private void collarToolStripMenuItem_Click(object sender, EventArgs e)
        {
            try
            {
                DataRow[] dato = dtFormsAllowed.Select("nombre_Real_Form = 'frmCollarAsign'");
                if (dato.Length > 0)
                {
                    frmCollarAsign oAsign = new frmCollarAsign();
                    oAsign.MdiParent = this;
                    oAsign.Show();
                }
                else
                {
                    MessageBox.Show("Form is not allowed", "Shipment", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error: " + ex.Message);
            }
        }

        private void passwordChangeToolStripMenuItem_Click(object sender, EventArgs e)
        {
            try
            {
                frmChangeLoggin oCh = new frmChangeLoggin();
                oCh.MdiParent = this;
                oCh.Show();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void reportTransactionToolStripMenuItem_Click(object sender, EventArgs e)
        {
            try
            {
                DataRow[] dato = dtFormsAllowed.Select("nombre_Real_Form = 'frmReportTrans'");
                if (dato.Length > 0)
                {
                    frmReportTrans oRpt = new frmReportTrans();
                    oRpt.MdiParent = this;
                    oRpt.Show();
                }
                else
                {
                    MessageBox.Show("Form is not allowed", "Shipment", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void validationLoggingToolStripMenuItem_Click(object sender, EventArgs e)
        {
            frmValidation oVal = new frmValidation();
            oVal.MdiParent = this;
            oVal.Show();
        }
    }
}

File: LogginColombiaGold/frmPpal.vb
Imports System.Windows.Forms
Imports System.Configuration

Public Class frmPpal
    Dim oRf As New clsRf
    Dim dtForms As New DataTable
    Private Sub ShowNewForm(ByVal sender As Object, ByVal e As EventArgs)
        ' Create a new instance of the child form.
        Dim ChildForm As New System.Windows.Forms.Form
        ' Make it a child of this MDI form before showing it.
        ChildForm.MdiParent = Me

        m_ChildFormNumber += 1
        ChildForm.Text = "Window " & m_ChildFormNumber

        ChildForm.Show()
    End Sub

    Private Sub OpenFile(ByVal sender As Object, ByVal e As EventArgs)
        Dim OpenFileDialog As New OpenFileDialog
        OpenFileDialog.InitialDirectory = My.Computer.FileSystem.SpecialDirectories.MyDocuments
        OpenFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"
        If (OpenFileDialog.ShowDialog(Me) = System.Windows.Forms.DialogResult.OK) Then
            Dim FileName As String = OpenFileDialog.FileName
            ' TODO: Add code here to open the file.
        End If
    End Sub

    Private Sub SaveAsToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Dim SaveFileDialog As New SaveFileDialog
        SaveFileDialog.InitialDirectory = My.Computer.FileSystem.SpecialDirectories.MyDocuments
        SaveFileDialog.Filter = "Text Files (*.txt)|*.txt|All Files (*.*)|*.*"

        If (SaveFileDialog.ShowDialog(Me) = System.Windows.Forms.DialogResult.OK) Then
            Dim FileName As String = SaveFileDialog.FileName
            ' TODO: Add code here to save the current contents of the form to a file.
        End If
    End Sub


    Private Sub ExitToolsStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.Close()
    End Sub

    Private Sub CutToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Use My.Computer.Clipboard to insert the selected text or images into the clipboard
    End Sub

    Private Sub CopyToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Use My.Computer.Clipboard to insert the selected text or images into the clipboard
    End Sub

    Private Sub PasteToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        'Use My.Computer.Clipboard.GetText() or My.Computer.Clipboard.GetData to retrieve information from the clipboard.
    End Sub


    Private Sub CascadeToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.Cascade)
    End Sub

    Private Sub TileVerticalToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.TileVertical)
    End Sub

    Private Sub TileHorizontalToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.TileHorizontal)
    End Sub

    Private Sub ArrangeIconsToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        Me.LayoutMdi(MdiLayout.ArrangeIcons)
    End Sub

    Private Sub CloseAllToolStripMenuItem_Click(ByVal sender As Object, ByVal e As EventArgs)
        ' Close all child forms of the parent.
        For Each ChildForm As Form In Me.MdiChildren
            ChildForm.Close()
        Next
    End Sub

    Private m_ChildFormNumber As Integer

    Private Sub frmPpal_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        oRf.dsPermisos = New DataSet()
        Application.Exit()
    End Sub

    Private Sub frmPpal_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.Show()

        'Cambia el color del fondo
        Dim ctlMDI As MdiClient
        Dim ctl As Control
        For Each ctl In Me.Controls
            Try
                ctlMDI = CType(ctl, MdiClient)
                ctlMDI.BackColor = Color.White
            Catch ex As InvalidCastException
            End Try
        Next

        dtForms = oRf.getFormsByGrupo(clsRf.sIdGrupo, ConfigurationSettings.AppSettings("IDProject").ToString)
        clsRf.dsPermisos = oRf.getFormsByGrupoAll(clsRf.sIdGrupo)

    End Sub

    Private Sub LogOutToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles LogOutToolStripMenuItem.Click
        Application.Exit()
    End Sub

    Private Sub DrillingToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DrillingToolStripMenuItem.Click
        frmDrilling.MdiParent = Me
        frmDrilling.Show()
    End Sub

    Private Sub ChangePasswordToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ChangePasswordToolStripMenuItem.Click
        frmChangeLoggin.MdiParent = Me
        frmChangeLoggin.Show()
    End Sub

    Private Sub SurveyToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles SurveyToolStripMenuItem.Click
        Dim drSeg As DataRow()
        drSeg = clsRf.dsPermisos.Tables(1).Select("nombre_Real_Form = 'frmSurvey' and Accion = 'Insertar'")
        If drSeg.Length > 0 Then

            frmSurvey.MdiParent = Me
            frmSurvey.Show()
        Else
            MsgBox("Not authorized to view this form", MsgBoxStyle.Exclamation, "Error")
        End If
    End Sub

    Private Sub TargetDrillingByCompanyToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        frmReporteMaquina.MdiParent = Me
        frmReporteMaquina.Show()
    End Sub

    Private Sub DrillingReportToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles DrillingReportToolStripMenuItem.Click
        frmDrillingReport.MdiParent = Me
        frmDrillingReport.Show()
    End Sub
End Class
